File: frontend/src/lib/api/packages.ts
import { postJson } from "@/lib/api/client";
import type { TravelPackageRequest, TravelPackageResponse } from "@/types/package";


export function generateTravelPackage(payload: TravelPackageRequest) {
  return postJson<TravelPackageResponse, TravelPackageRequest>(
    "/api/v1/packages/generate",
    payload,
  );
}

File: backend/app/api/routes/packages.py
from fastapi import APIRouter

from app.schemas.package import TravelPackageRequest, TravelPackageResponse
from app.services.package_generator import generate_travel_package


router = APIRouter(prefix="/packages", tags=["packages"])


@router.post("/generate", response_model=TravelPackageResponse)
def generate_package(
    payload: TravelPackageRequest,
) -> TravelPackageResponse:
    return generate_travel_package(payload)
